2.1 installation problems

I'm installing version 2.1. Step 4 of upgrade is done, but desktop and mobile applications do not work : 

https://preprod.cartoriviera.ch/v2/
https://preprod.cartoriviera.ch/v2/mobile/

I still have the following error in step 4 : 

```
No handlers could be found for logger "c2cgeoportal"
ERROR: The database is probably not up to date
(psycopg2.ProgrammingError) column layer_wms.ogc_server_id does not exist
LINE 1: ...lude_properties AS main_layer_exclude_properties, main.layer...
                                                             ^
 [SQL: 'SELECT main.treeitem.type AS main_treeitem_type, main.layer_wms.id AS main_layer_wms_id, main.layer.id AS main_layer_id, main.treeitem.id AS main_treeitem_id, main.treeitem.name AS main_treeitem_name, main.treeitem.metadata_url AS main_treeitem_metadata_url, main.treeitem.description AS main_treeitem_description, main.layer.public AS main_layer_public, main.layer.geo_table AS main_layer_geo_table, main.layer.exclude_properties AS main_layer_exclude_properties, main.layer_wms.ogc_server_id AS main_layer_wms_ogc_server_id, main.layer_wms.layer AS main_layer_wms_layer, main.layer_wms.style AS main_layer_wms_style, main.layer_wms.time_mode AS main_layer_wms_time_mode, main.layer_wms.time_widget AS main_layer_wms_time_widget \nFROM main.treeitem JOIN main.layer ON main.treeitem.id = main.layer.id JOIN main.layer_wms ON main.layer.id = main.layer_wms.id']
# removes the always changed date line
```

Here is my step 4 log

find: `cartoriviera/static-ngeo/components': No such file or directory
find: `cartoriviera/static-ngeo/components': No such file or directory
find: `cartoriviera/static-ngeo/components': No such file or directory
CONST_Makefile:1112: warning: overriding recipe for target 'apache/mapcache.xml'
cartoriviera.mk:17: warning: ignoring old recipe for target 'apache/mapcache.xml'
.build/venv/bin/c2ctool upgrade v2.mk  --step 4
make[1]: Entering directory '/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish'
find: `cartoriviera/static-ngeo/components': No such file or directory
find: `cartoriviera/static-ngeo/components': No such file or directory
find: `cartoriviera/static-ngeo/components': No such file or directory
CONST_Makefile:1112: warning: overriding recipe for target 'apache/mapcache.xml'
cartoriviera.mk:17: warning: ignoring old recipe for target 'apache/mapcache.xml'
.build/venv/bin/c2c-template --vars vars_prod_v2.yaml --get-config .build/config.yaml lidar_http lidar_https targetemail servicesenderemail emailhost instanceid sqlalchemy.url schema parentschema enable_admin_interface pyramid_closure node_modules_path closure_library_path default_locale_name servers layers available_locale_names cache admin_interface functionalities external_themes_url raster shortener hide_capabilities mapserverproxy tinyowsproxy resourceproxy print_url tiles_url checker check_collector default_max_age jsbuild package srid reset_password fulltextsearch headers authorized_referers hooks stats db_chooser
find: `cartoriviera/static-ngeo/components': No such file or directory
.build/venv/bin/pot-create --config lingua-client.cfg --output cartoriviera/locale/cartoriviera-client.pot cartoriviera/templates/mobile.html cartoriviera/templates/index.html cartoriviera/templates/api/xapihelp.html cartoriviera/templates/api/apihelp.html cartoriviera/templates/search-tip.html cartoriviera/templates/edit.html cartoriviera/templates/apihelp.html cartoriviera/templates/routing.html cartoriviera/static-ngeo/js/cartorivieramodule.js cartoriviera/static-ngeo/js/mobile.js  .build/config.yaml development.ini
Error parsing ./cartoriviera/templates/mobile.html at line 6 column 74
Error parsing ./cartoriviera/templates/api/xapihelp.html at line 15 column 45
Error parsing ./cartoriviera/templates/api/apihelp.html at line 15 column 45
No handlers could be found for logger "c2cgeoportal"
ERROR: The database is probably not up to date
(psycopg2.ProgrammingError) column layer_wms.ogc_server_id does not exist
LINE 1: ...lude_properties AS main_layer_exclude_properties, main.layer...
                                                             ^
 [SQL: 'SELECT main.treeitem.type AS main_treeitem_type, main.layer_wms.id AS main_layer_wms_id, main.layer.id AS main_layer_id, main.treeitem.id AS main_treeitem_id, main.treeitem.name AS main_treeitem_name, main.treeitem.metadata_url AS main_treeitem_metadata_url, main.treeitem.description AS main_treeitem_description, main.layer.public AS main_layer_public, main.layer.geo_table AS main_layer_geo_table, main.layer.exclude_properties AS main_layer_exclude_properties, main.layer_wms.ogc_server_id AS main_layer_wms_ogc_server_id, main.layer_wms.layer AS main_layer_wms_layer, main.layer_wms.style AS main_layer_wms_style, main.layer_wms.time_mode AS main_layer_wms_time_mode, main.layer_wms.time_widget AS main_layer_wms_time_widget \nFROM main.treeitem JOIN main.layer ON main.treeitem.id = main.layer.id JOIN main.layer_wms ON main.layer.id = main.layer_wms.id']
# removes the always changed date line
sed -i '/^"POT-Creation-Date: /d' cartoriviera/locale/cartoriviera-client.pot
sed -i '/^"PO-Revision-Date: /d' cartoriviera/locale/cartoriviera-client.pot
chmod go+r cartoriviera/locale/cartoriviera-client.pot
mkdir -p cartoriviera/locale/fr/LC_MESSAGES/
[ -f cartoriviera/locale/fr/LC_MESSAGES/cartoriviera-client.po ] || msginit --no-translator --input=cartoriviera/locale/cartoriviera-client.pot --output-file=cartoriviera/locale/fr/LC_MESSAGES/cartoriviera-client.po -l fr
touch --no-create cartoriviera/locale/fr/LC_MESSAGES/cartoriviera-client.po
msgmerge --backup=none --update --sort-output --no-location cartoriviera/locale/fr/LC_MESSAGES/cartoriviera-client.po cartoriviera/locale/cartoriviera-client.pot
.................. done.
mkdir -p .build/locale/fr/LC_MESSAGES/
.build/venv/bin/tx pull -l fr --force
Pulling translations for resource ngeo.gmf-2_1 (source: .build/locale/gmf.pot)
Skipping 'fr' translation (file: .build/locale/fr/LC_MESSAGES/gmf.po).
Done.
touch --date "2016-03-30 10:59:50.167252472 +0200" /home/dgnerre/.transifexrc
mkdir -p cartoriviera/static-ngeo/build/
node node_modules/.bin/compile-catalog .build/locale/fr/LC_MESSAGES/gmf.po cartoriviera/locale/fr/LC_MESSAGES/cartoriviera-client.po > cartoriviera/static-ngeo/build/fr.json
mkdir -p .build/externs/
curl --max-redirs 0 --location --output .build/externs/angular-1.5.js https://raw.githubusercontent.com/google/closure-compiler/b8d8d1d6c6d245a3df57feedc511f2e849b01932/contrib/externs/angular-1.5.js
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 62129  100 62129    0     0   224k      0 --:--:-- --:--:-- --:--:--  225k
node node_modules/ngeo/buildtools/build.js .build/desktop.json .build/desktop.js
info ol Parsing dependencies
info ol Compiling 461 sources
mkdir -p cartoriviera/static-ngeo/build/
awk 'FNR==1{print ""}1' node_modules/jquery/dist/jquery.min.js node_modules/ngeo/third-party/jquery-ui/jquery-ui.min.js node_modules/angular/angular.min.js node_modules/angular-animate/angular-animate.min.js node_modules/angular-sanitize/angular-sanitize.min.js node_modules/angular-touch/angular-touch.min.js node_modules/typeahead.js/dist/typeahead.bundle.min.js node_modules/angular-ui-date/dist/date.js node_modules/angular-float-thead/angular-floatThead.js node_modules/floatthead/dist/jquery.floatThead.min.js node_modules/angular-ui-slider/src/slider.js node_modules/proj4/dist/proj4.js node_modules/angular-gettext/dist/angular-gettext.min.js node_modules/angular-dynamic-locale/dist/tmhDynamicLocale.min.js node_modules/bootstrap/dist/js/bootstrap.min.js node_modules/d3/d3.min.js  .build/desktop.js > cartoriviera/static-ngeo/build/desktop.js
sed -i '/^\/\/# sourceMappingURL=.*\.map$/d' cartoriviera/static-ngeo/build/desktop.js
PYTHONIOENCODING=UTF-8 .build/venv/bin/mako-render \
        --var 'src=cartoriviera/static-ngeo/js/**/*.js' \
        --var 'src_components=cartoriviera/static-ngeo/components/**/*.js' \
        --var src_set=contribs_gmf \
        --var entry_point=cartoriviera_mobile \
        --var ngeo_folder=node_modules/ngeo/ \
        --var generate_exports=true node_modules/ngeo/buildtools/mako_build.json > .build/mobile.json
node node_modules/ngeo/buildtools/build.js .build/mobile.json .build/mobile.js
info ol Parsing dependencies
info ol Compiling 461 sources
mkdir -p cartoriviera/static-ngeo/build/
awk 'FNR==1{print ""}1' node_modules/jquery/dist/jquery.min.js node_modules/ngeo/third-party/jquery-ui/jquery-ui.min.js node_modules/angular/angular.min.js node_modules/angular-animate/angular-animate.min.js node_modules/angular-sanitize/angular-sanitize.min.js node_modules/angular-touch/angular-touch.min.js node_modules/typeahead.js/dist/typeahead.bundle.min.js node_modules/angular-ui-date/dist/date.js node_modules/angular-float-thead/angular-floatThead.js node_modules/floatthead/dist/jquery.floatThead.min.js node_modules/angular-ui-slider/src/slider.js node_modules/proj4/dist/proj4.js node_modules/angular-gettext/dist/angular-gettext.min.js node_modules/angular-dynamic-locale/dist/tmhDynamicLocale.min.js node_modules/bootstrap/dist/js/bootstrap.min.js node_modules/d3/d3.min.js  .build/mobile.js > cartoriviera/static-ngeo/build/mobile.js
sed -i '/^\/\/# sourceMappingURL=.*\.map$/d' cartoriviera/static-ngeo/build/mobile.js
make[1]: *** No rule to make target 'cartoriviera/static-ngeo/build/desktop.css', needed by 'build-ngeo'.  Stop.
make[1]: Leaving directory '/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish'
Traceback (most recent call last):
  File ".build/venv/bin/c2ctool", line 11, in <module>
    sys.exit(main())
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/c2cgeoportal/scripts/c2ctool.py", line 98, in main
    c2ctool.upgrade()
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/c2cgeoportal/scripts/c2ctool.py", line 222, in upgrade
    self.step4()
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/c2cgeoportal/scripts/c2ctool.py", line 433, in step4
    check_call(["make", "-f", self.options.file, "build"])
  File "/usr/lib/python2.7/subprocess.py", line 540, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['make', '-f', 'v2.mk', 'build']' returned non-zero exit status 2
CONST_Makefile:1218: recipe for target 'upgrade4' failed
make: *** [upgrade4] Error 1
You have new mail in /var/mail/dgnerre
dgnerre@cartoriviera3:/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish$
You have new mail in /var/mail/dgnerre
dgnerre@cartoriviera3:/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish$
dgnerre@cartoriviera3:/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish$
dgnerre@cartoriviera3:/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish$
dgnerre@cartoriviera3:/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish$
dgnerre@cartoriviera3:/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish$ make -f v2.mk upgrade4
find: `cartoriviera/static-ngeo/components': No such file or directory
find: `cartoriviera/static-ngeo/components': No such file or directory
find: `cartoriviera/static-ngeo/components': No such file or directory
CONST_Makefile:1113: warning: overriding recipe for target 'apache/mapcache.xml'
cartoriviera.mk:17: warning: ignoring old recipe for target 'apache/mapcache.xml'
.build/venv/bin/c2ctool upgrade v2.mk  --step 4
make[1]: Entering directory '/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish'
find: `cartoriviera/static-ngeo/components': No such file or directory
find: `cartoriviera/static-ngeo/components': No such file or directory
find: `cartoriviera/static-ngeo/components': No such file or directory
CONST_Makefile:1113: warning: overriding recipe for target 'apache/mapcache.xml'
cartoriviera.mk:17: warning: ignoring old recipe for target 'apache/mapcache.xml'
.build/venv/bin/c2c-template --vars vars_prod_v2.yaml --engine mako --files jsbuild/app.cfg.mako
mkdir -p cartoriviera/static/build/
.build/venv/bin/jsbuild jsbuild/app.cfg --uncompress -j main.js -o cartoriviera/static/build
Writing to cartoriviera/static/build/main.js (4960 KB)
Done:
cartoriviera/static/build/main.js
mkdir -p cartoriviera/static/build/
.build/venv/bin/jsbuild jsbuild/app.cfg --uncompress -j edit.js -o cartoriviera/static/build
Writing to cartoriviera/static/build/edit.js (4889 KB)
Done:
cartoriviera/static/build/edit.js
mkdir -p cartoriviera/static/build/
.build/venv/bin/jsbuild jsbuild/app.cfg --uncompress -j routing.js -o cartoriviera/static/build
Writing to cartoriviera/static/build/routing.js (2996 KB)
Done:
cartoriviera/static/build/routing.js
mkdir -p cartoriviera/static/build/
.build/venv/bin/jsbuild jsbuild/app.cfg --uncompress -j api.js -o cartoriviera/static/build
Writing to cartoriviera/static/build/api.js (1441 KB)
Done:
cartoriviera/static/build/api.js
mkdir -p cartoriviera/static/build/
.build/venv/bin/jsbuild jsbuild/app.cfg --uncompress -j xapi.js -o cartoriviera/static/build
Writing to cartoriviera/static/build/xapi.js (3772 KB)
Done:
cartoriviera/static/build/xapi.js
mkdir -p cartoriviera/static/build/
.build/venv/bin/jsbuild jsbuild/app.cfg --uncompress -j lang-fr.js -o cartoriviera/static/build
Writing to cartoriviera/static/build/lang-fr.js (110 KB)
Done:
cartoriviera/static/build/lang-fr.js
mkdir -p cartoriviera/static/build/
.build/venv/bin/jsbuild jsbuild/app.cfg --uncompress -j api-lang-fr.js -o cartoriviera/static/build
Writing to cartoriviera/static/build/api-lang-fr.js (63 KB)
Done:
cartoriviera/static/build/api-lang-fr.js
.build/venv/bin/c2c-template --vars vars_prod_v2.yaml --get-config .build/config.yaml lidar_http lidar_https targetemail servicesenderemail emailhost instanceid sqlalchemy.url schema parentschema enable_admin_interface pyramid_closure node_modules_path closure_library_path default_locale_name servers layers available_locale_names cache admin_interface functionalities external_themes_url raster shortener hide_capabilities mapserverproxy tinyowsproxy resourceproxy print_url tiles_url checker check_collector default_max_age jsbuild package srid reset_password fulltextsearch headers authorized_referers hooks stats db_chooser
find: `cartoriviera/static-ngeo/components': No such file or directory
.build/venv/bin/pot-create --config lingua-client.cfg --output cartoriviera/locale/cartoriviera-client.pot cartoriviera/templates/mobile.html cartoriviera/templates/index.html cartoriviera/templates/api/xapihelp.html cartoriviera/templates/api/apihelp.html cartoriviera/templates/search-tip.html cartoriviera/templates/edit.html cartoriviera/templates/apihelp.html cartoriviera/templates/routing.html cartoriviera/static-ngeo/js/cartorivieramodule.js cartoriviera/static-ngeo/js/mobile.js  .build/config.yaml development.ini
Error parsing ./cartoriviera/templates/mobile.html at line 6 column 74
Error parsing ./cartoriviera/templates/api/xapihelp.html at line 15 column 45
Error parsing ./cartoriviera/templates/api/apihelp.html at line 15 column 45
No handlers could be found for logger "c2cgeoportal"
ERROR: The database is probably not up to date
(psycopg2.ProgrammingError) column layer_wms.ogc_server_id does not exist
LINE 1: ...lude_properties AS main_layer_exclude_properties, main.layer...
                                                             ^
 [SQL: 'SELECT main.treeitem.type AS main_treeitem_type, main.layer_wms.id AS main_layer_wms_id, main.layer.id AS main_layer_id, main.treeitem.id AS main_treeitem_id, main.treeitem.name AS main_treeitem_name, main.treeitem.metadata_url AS main_treeitem_metadata_url, main.treeitem.description AS main_treeitem_description, main.layer.public AS main_layer_public, main.layer.geo_table AS main_layer_geo_table, main.layer.exclude_properties AS main_layer_exclude_properties, main.layer_wms.ogc_server_id AS main_layer_wms_ogc_server_id, main.layer_wms.layer AS main_layer_wms_layer, main.layer_wms.style AS main_layer_wms_style, main.layer_wms.time_mode AS main_layer_wms_time_mode, main.layer_wms.time_widget AS main_layer_wms_time_widget \nFROM main.treeitem JOIN main.layer ON main.treeitem.id = main.layer.id JOIN main.layer_wms ON main.layer.id = main.layer_wms.id']
# removes the always changed date line
sed -i '/^"POT-Creation-Date: /d' cartoriviera/locale/cartoriviera-client.pot
sed -i '/^"PO-Revision-Date: /d' cartoriviera/locale/cartoriviera-client.pot
chmod go+r cartoriviera/locale/cartoriviera-client.pot
mkdir -p cartoriviera/locale/fr/LC_MESSAGES/
[ -f cartoriviera/locale/fr/LC_MESSAGES/cartoriviera-client.po ] || msginit --no-translator --input=cartoriviera/locale/cartoriviera-client.pot --output-file=cartoriviera/locale/fr/LC_MESSAGES/cartoriviera-client.po -l fr
touch --no-create cartoriviera/locale/fr/LC_MESSAGES/cartoriviera-client.po
msgmerge --backup=none --update --sort-output --no-location cartoriviera/locale/fr/LC_MESSAGES/cartoriviera-client.po cartoriviera/locale/cartoriviera-client.pot
.................. done.
mkdir -p .build/locale/fr/LC_MESSAGES/
.build/venv/bin/tx pull -l fr --force
Pulling translations for resource ngeo.gmf-2_1 (source: .build/locale/gmf.pot)
Skipping 'fr' translation (file: .build/locale/fr/LC_MESSAGES/gmf.po).
Done.
touch --date "2016-03-30 10:59:50.167252472 +0200" /home/dgnerre/.transifexrc
mkdir -p cartoriviera/static-ngeo/build/
node node_modules/.bin/compile-catalog .build/locale/fr/LC_MESSAGES/gmf.po cartoriviera/locale/fr/LC_MESSAGES/cartoriviera-client.po > cartoriviera/static-ngeo/build/fr.json
touch --no-create node_modules/ngeo/contribs/gmf/fonts/gmf-icons.svg
node_modules/.bin/svg2ttf node_modules/ngeo/contribs/gmf/fonts/gmf-icons.svg node_modules/ngeo/contribs/gmf/fonts/gmf-icons.ttf
node_modules/.bin/ttf2woff node_modules/ngeo/contribs/gmf/fonts/gmf-icons.ttf node_modules/ngeo/contribs/gmf/fonts/gmf-icons.woff
node_modules/.bin/ttf2eot node_modules/ngeo/contribs/gmf/fonts/gmf-icons.ttf node_modules/ngeo/contribs/gmf/fonts/gmf-icons.eot
mkdir -p cartoriviera/static-ngeo/fonts/
cp node_modules/ngeo/contribs/gmf/fonts/gmf-icons.woff cartoriviera/static-ngeo/fonts/gmf-icons.woff
mkdir -p cartoriviera/static-ngeo/fonts/
cp node_modules/ngeo/contribs/gmf/fonts/gmf-icons.ttf cartoriviera/static-ngeo/fonts/gmf-icons.ttf
mkdir -p cartoriviera/static-ngeo/fonts/
cp node_modules/ngeo/contribs/gmf/fonts/gmf-icons.eot cartoriviera/static-ngeo/fonts/gmf-icons.eot
mkdir -p cartoriviera/static-ngeo/fonts/
cp node_modules/font-awesome/fonts/fontawesome-webfont.woff cartoriviera/static-ngeo/fonts/fontawesome-webfont.woff
mkdir -p cartoriviera/static-ngeo/fonts/
cp node_modules/font-awesome/fonts/fontawesome-webfont.woff2 cartoriviera/static-ngeo/fonts/fontawesome-webfont.woff2
mkdir -p cartoriviera/static-ngeo/fonts/
cp node_modules/font-awesome/fonts/fontawesome-webfont.ttf cartoriviera/static-ngeo/fonts/fontawesome-webfont.ttf
mkdir -p cartoriviera/static-ngeo/fonts/
cp node_modules/font-awesome/fonts/fontawesome-webfont.eot cartoriviera/static-ngeo/fonts/fontawesome-webfont.eot
mkdir -p cartoriviera/static-ngeo/build/
node_modules/.bin/lessc --autoprefix cartoriviera/static-ngeo/less/mobile-build.less cartoriviera/static-ngeo/build/mobile.css
mkdir -p cartoriviera/static-ngeo/build/
wget -O cartoriviera/static-ngeo/build/angular-locale_fr.js https://raw.githubusercontent.com/angular/angular.js/master/src/ngLocale/angular-locale_fr.js
--2016-09-15 07:45:21--  https://raw.githubusercontent.com/angular/angular.js/master/src/ngLocale/angular-locale_fr.js
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 151.101.12.133
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|151.101.12.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 2453 (2.4K) [text/plain]
Saving to: âcartoriviera/static-ngeo/build/angular-locale_fr.jsâ

cartoriviera/static-ngeo/build/angular-locale_f 100%[========================================================================================================>]   2.40K  --.-KB/s   in 0s

2016-09-15 07:45:21 (60.0 MB/s) - âcartoriviera/static-ngeo/build/angular-locale_fr.jsâ saved [2453/2453]

.build/venv/bin/c2c-template --vars vars_prod_v2.yaml --engine mako --files .whiskey/action_hooks/pre-build.mako
.build/venv/bin/c2c-template --vars vars_prod_v2.yaml --engine mako --files print/config.yaml.mako
.build/venv/bin/c2c-template --vars vars_prod_v2.yaml --engine mako --files print/WEB-INF/classes/logback.xml.mako
.build/venv/bin/c2c-template --vars vars_prod_v2.yaml --engine mako --files deploy/deploy.cfg.mako
.build/venv/bin/c2c-template --vars vars_prod_v2.yaml --engine mako --files deploy/hooks/pre-restore-database.mako
.build/venv/bin/c2c-template --vars vars_prod_v2.yaml --engine mako --files apache/frontend.conf.mako
.build/venv/bin/c2c-template --vars vars_prod_v2.yaml --engine mako --files apache/README.txt.mako
.build/venv/bin/c2c-template --vars vars_prod_v2.yaml --engine mako --files apache/application.wsgi.mako
.build/venv/bin/c2c-template --vars vars_prod_v2.yaml --engine mako --files apache/wsgi.conf.mako
.build/venv/bin/c2c-template --vars vars_prod_v2.yaml --engine mako --files apache/php-fpm.conf.mako
.build/venv/bin/c2c-template --vars vars_prod_v2.yaml --engine mako --files testDB/11-schemas.sql.mako
.build/venv/bin/c2c-template --vars vars_prod_v2.yaml --engine mako --files testDB/01-schemas.sql.mako
.build/venv/bin/c2c-template --vars vars_prod_v2.yaml --engine mako --files testDB/Dockerfile.mako
.build/venv/bin/c2c-template --vars vars_prod_v2.yaml --engine mako --files production.ini.mako
.build/venv/bin/c2c-template --vars vars_prod_v2.yaml --engine mako --files mapserver/replace_db.sh.mako
.build/venv/bin/c2c-template --vars vars_prod_v2.yaml --engine mako --files logback.xml.mako
.build/venv/bin/c2c-template --vars vars_prod_v2.yaml --engine mako --files CONST_create_template/jsbuild/app.cfg.mako
.build/venv/bin/c2c-template --vars vars_prod_v2.yaml --engine mako --files CONST_create_template/print/WEB-INF/classes/logback.xml.mako
.build/venv/bin/c2c-template --vars vars_prod_v2.yaml --engine mako --files CONST_create_template/print/print-apps/cartoriviera/config.yaml.mako
.build/venv/bin/c2c-template --vars vars_prod_v2.yaml --engine mako --files CONST_create_template/deploy/deploy.cfg.mako
.build/venv/bin/c2c-template --vars vars_prod_v2.yaml --engine mako --files CONST_create_template/deploy/hooks/pre-restore-database.mako
.build/venv/bin/c2c-template --vars vars_prod_v2.yaml --engine mako --files CONST_create_template/apache/frontend.conf.mako
.build/venv/bin/c2c-template --vars vars_prod_v2.yaml --engine mako --files CONST_create_template/apache/mapserver.conf.mako
.build/venv/bin/c2c-template --vars vars_prod_v2.yaml --engine mako --files CONST_create_template/apache/README.txt.mako
.build/venv/bin/c2c-template --vars vars_prod_v2.yaml --engine mako --files CONST_create_template/apache/application.wsgi.mako
.build/venv/bin/c2c-template --vars vars_prod_v2.yaml --engine mako --files CONST_create_template/apache/wsgi.conf.mako
.build/venv/bin/c2c-template --vars vars_prod_v2.yaml --engine mako --files CONST_create_template/apache/tinyows.conf.mako
.build/venv/bin/c2c-template --vars vars_prod_v2.yaml --engine mako --files CONST_create_template/production.ini.mako
.build/venv/bin/c2c-template --vars vars_prod_v2.yaml --engine mako --files CONST_create_template/mapserver/demo.map.mako
.build/venv/bin/c2c-template --vars vars_prod_v2.yaml --engine mako --files CONST_create_template/mapserver/tinyows.xml.mako
.build/venv/bin/c2c-template --vars vars_prod_v2.yaml --engine mako --files CONST_create_template/mapserver/c2cgeoportal.map.mako
.build/venv/bin/c2c-template --vars vars_prod_v2.yaml --engine mako --files CONST_create_template/alembic.ini.mako
.build/venv/bin/c2c-template --vars vars_prod_v2.yaml --engine mako --files CONST_create_template/development.ini.mako
.build/venv/bin/c2c-template --vars vars_prod_v2.yaml --engine mako --files CONST_create_template/project.yaml.mako
.build/venv/bin/c2c-template --vars vars_prod_v2.yaml --engine mako --files CONST_create_template/alembic_static.ini.mako
.build/venv/bin/c2c-template --vars vars_prod_v2.yaml --engine mako --files CONST_create_template/docker-compose.yml.mako
.build/venv/bin/pot-create --keyword _ --config lingua-server.cfg --output cartoriviera/locale/cartoriviera-server.pot cartoriviera/models.py cartoriviera/templates/mobile.html cartoriviera/templates/index.html cartoriviera/templates/api/xapihelp.html cartoriviera/templates/api/apihelp.html cartoriviera/templates/search-tip.html cartoriviera/templates/edit.html cartoriviera/templates/apihelp.html cartoriviera/templates/routing.html
/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/lingua/extract.py:327: UnicodeWarning: Python extractor called with bytes input. Please update your plugin to submit unicode instead.
  for message in extractor(real_filename, options):
# removes the always changed date line
sed -i '/^"POT-Creation-Date: /d' cartoriviera/locale/cartoriviera-server.pot
sed -i '/^"PO-Revision-Date: /d' cartoriviera/locale/cartoriviera-server.pot
chmod go+r cartoriviera/locale/cartoriviera-server.pot
mkdir -p cartoriviera/locale/fr/LC_MESSAGES/
[ -f cartoriviera/locale/fr/LC_MESSAGES/cartoriviera-server.po ] || msginit --no-translator --input=cartoriviera/locale/cartoriviera-server.pot --output-file=cartoriviera/locale/fr/LC_MESSAGES/cartoriviera-server.po -l fr
touch --no-create cartoriviera/locale/fr/LC_MESSAGES/cartoriviera-server.po
msgmerge --backup=none --update --sort-output --no-location cartoriviera/locale/fr/LC_MESSAGES/cartoriviera-server.po cartoriviera/locale/cartoriviera-server.pot
... done.
msgfmt -o cartoriviera/locale/fr/LC_MESSAGES/cartoriviera-server.mo cartoriviera/locale/fr/LC_MESSAGES/cartoriviera-server.po
touch --no-create cartoriviera/locale/fr/LC_MESSAGES/cartoriviera-server.mo
.build/venv/bin/generate_controller --generate-apache-config
sudo /usr/sbin/apache2ctl graceful
touch .build/apache.timestamp
.build/venv/bin/c2c-versions CONST_packages.yaml common main print tilecloud-chain deploy ngeo
Usage: {cmd} <config file> <section>
touch .build/test-packages.timestamp
make[1]: Leaving directory '/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish'
INFO  [alembic.runtime.migration] Context impl PostgresqlImpl.
INFO  [alembic.runtime.migration] Will assume transactional DDL.
INFO  [alembic.runtime.migration] Running upgrade 29f2a32859ec -> 951ff84bd8ec, Be able to delete a WMS layer in SQL
INFO  [alembic.runtime.migration] Running upgrade 29f2a32859ec -> 6a412d9437b1, Rename ServerOGC to OGCServer
INFO  [alembic.runtime.migration] Running upgrade 6a412d9437b1, 951ff84bd8ec -> daf738d5bae4, Merge 2.0 and master branches
INFO  [alembic.runtime.migration] Context impl PostgresqlImpl.
INFO  [alembic.runtime.migration] Will assume transactional DDL.

================================================================
The upgrade is nearly done, now you should:
- Test your application.
To continue type:
make -f v2.mk upgrade5

I get now this error... : 

No handlers could be found for logger "c2cgeoportal"
Get DescribeFeatureType for url: http://localhost/v2/mapserv?VERSION=1.1.0&REQUEST=DescribeFeatureType&SERVICE=WFS
Get DescribeFeatureType for url: https://www.cartoriviera.ch/qgis/qgis_mapserver.fcgi?MAP=/usr/local/share/qgis/signalisation.qgs&transparent=true&VERSION=1.1.0&REQUEST=DescribeFeatureType&SERVICE=WFS
Get DescribeFeatureType for url: /qgis/qgis_mapserver.fcgi?map=/var/sig/qgis/distribution.qgs&transparent=true&VERSION=1.1.0&REQUEST=DescribeFeatureType&SERVICE=WFS
Unable to DescribeFeatureType from URL /qgis/qgis_mapserver.fcgi?map=/var/sig/qgis/distribution.qgs&transparent=true&VERSION=1.1.0&REQUEST=DescribeFeatureType&SERVICE=WFS
No such table '' for layer 'separateurs_hydro'.
CONST_Makefile:676: recipe for target 'cartoriviera/locale/cartoriviera-client.pot' failed
make[1]: *** [cartoriviera/locale/cartoriviera-client.pot] Error 1
make[1]: Leaving directory '/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish'
Traceback (most recent call last):
  File ".build/venv/bin/c2ctool", line 11, in <module>
    sys.exit(main())
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/c2cgeoportal/scripts/c2ctool.py", line 98, in main
    c2ctool.upgrade()
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/c2cgeoportal/scripts/c2ctool.py", line 222, in upgrade
    self.step4()
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/c2cgeoportal/scripts/c2ctool.py", line 433, in step4
    check_call(["make", "-f", self.options.file, "build"])
  File "/usr/lib/python2.7/subprocess.py", line 540, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['make', '-f', 'v2.mk', 'build']' returned non-zero exit status 2
CONST_Makefile:1219: recipe for target 'upgrade4' failed
make: *** [upgrade4] Error 1

I tried to install 2.1RC1, and I have the following error in step 4

CONST_Makefile:1131: warning: overriding recipe for target 'apache/mapcache.xml'
cartoriviera.mk:17: warning: ignoring old recipe for target 'apache/mapcache.xml'
.build/venv/bin/c2c-template --vars vars_prod_v2.yaml --engine mako --files tilegeneration/config.yaml.mako
Traceback (most recent call last):
  File ".build/venv/bin/c2c-template", line 11, in <module>
    sys.exit(main())
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/c2c/template.py", line 213, in main
    _proceed(files, used_vars, options)
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/c2c/template.py", line 237, in _proceed
    bottle_template(files, used_vars, engine)
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/c2c/template.py", line 268, in bottle_template
    template, **used_vars
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/bin/bottle.py", line 3609, in template
    return TEMPLATES[tplid].render(kwargs)
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/bin/bottle.py", line 3280, in render
    return self.tpl.render(**_defaults)
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/mako/template.py", line 445, in render
    return runtime._render(self, self.callable_, args, data)
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/mako/runtime.py", line 829, in _render
    **_kwargs_for_callable(callable_, data))
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/mako/runtime.py", line 864, in _render_context
    _exec_template(inherit, lclcontext, args=args, kwargs=kwargs)
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/mako/runtime.py", line 890, in _exec_template
    callable_(context, *args, **kwargs)
  File "tilegeneration_config_yaml_mako", line 99, in render_body
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/mako/runtime.py", line 226, in __str__
    raise NameError("Undefined")
NameError: Undefined
CONST_Makefile:635: recipe for target 'tilegeneration/config.yaml' failed
make[1]: *** [tilegeneration/config.yaml] Error 1
make[1]: Leaving directory '/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish'
Traceback (most recent call last):
  File ".build/venv/bin/c2ctool", line 11, in <module>
    sys.exit(main())
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/c2cgeoportal/scripts/c2ctool.py", line 98, in main
    c2ctool.upgrade()
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/c2cgeoportal/scripts/c2ctool.py", line 222, in upgrade
    self.step4()
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/c2cgeoportal/scripts/c2ctool.py", line 436, in step4
    check_call(["make", "-f", self.options.file, "build"])
  File "/usr/lib/python2.7/subprocess.py", line 540, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['make', '-f', 'v2.mk', 'build']' returned non-zero exit status 2
CONST_Makefile:1237: recipe for target 'upgrade4' failed
make: *** [upgrade4] Error 1

Ok thanks, it was a misundestanding with the changes to apply. Now I haven't this error anymore, but I have this one : 
 - /layers/rayonnement_solaire_2012: Cannot find required key 'extension'.
 - /layers/rayonnement_solaire_2012: Cannot find required key 'mime_type'.
 - /layers/rayonnement_solaire_2012: Cannot find required key 'wmts_style'.
 - /layers/vevey_moyen_age: Cannot find required key 'wmts_style'.
CONST_Makefile:1131: recipe for target 'apache/mapcache.xml' failed
make[1]: *** [apache/mapcache.xml] Error 1
make[1]: Leaving directory '/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish'
Traceback (most recent call last):
  File ".build/venv/bin/c2ctool", line 11, in <module>
    sys.exit(main())
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/c2cgeoportal/scripts/c2ctool.py", line 98, in main
    c2ctool.upgrade()
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/c2cgeoportal/scripts/c2ctool.py", line 222, in upgrade
    self.step4()
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/c2cgeoportal/scripts/c2ctool.py", line 436, in step4
    check_call(["make", "-f", self.options.file, "build"])
  File "/usr/lib/python2.7/subprocess.py", line 540, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['make', '-f', 'v2.mk', 'build']' returned non-zero exit status 2
CONST_Makefile:1237: recipe for target 'upgrade4' failed
make: *** [upgrade4] Error 1

I juste tested, now I get this error

find: `cartoriviera/static-ngeo/components': No such file or directory
find: `cartoriviera/static-ngeo/components': No such file or directory
find: `cartoriviera/static-ngeo/components': No such file or directory
CONST_Makefile:1131: warning: overriding recipe for target 'apache/mapcache.xml'
cartoriviera.mk:17: warning: ignoring old recipe for target 'apache/mapcache.xml'
.build/venv/bin/c2c-template --vars vars_prod_v2.yaml --engine mako --files tilegeneration/config.yaml.mako
.build/venv/bin/generate_controller --generate-mapcache-config
ERROR:tilecloud_chain:__init__:The config file 'tilegeneration/config.yaml' in invalid.
 - /generation: Key 'disable_code' was not defined.
 - /generation: Key 'disable_database' was not defined.
 - /generation: Key 'disable_fillqueue' was not defined.
 - /generation: Key 'disable_geodata' was not defined.
 - /generation: Key 'disable_tilesgen' was not defined.
CONST_Makefile:1131: recipe for target 'apache/mapcache.xml' failed
make[1]: *** [apache/mapcache.xml] Error 1
make[1]: Leaving directory '/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish'
Traceback (most recent call last):
  File ".build/venv/bin/c2ctool", line 11, in <module>
    sys.exit(main())
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/c2cgeoportal/scripts/c2ctool.py", line 98, in main
    c2ctool.upgrade()
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/c2cgeoportal/scripts/c2ctool.py", line 222, in upgrade
    self.step4()
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/c2cgeoportal/scripts/c2ctool.py", line 436, in step4
    check_call(["make", "-f", self.options.file, "build"])
  File "/usr/lib/python2.7/subprocess.py", line 540, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['make', '-f', 'v2.mk', 'build']' returned non-zero exit status 2
CONST_Makefile:1237: recipe for target 'upgrade4' failed
make: *** [upgrade4] Error 1

Ok it goes further, but now I have the following errors (one in the middle and one at the end of step) : 

........
Get DescribeFeatureType for url: /qgis/qgis_mapserver.fcgi?map=/var/sig/qgis/distribution.qgs&transparent=true&VERSION=1.1.0&REQUEST=DescribeFeatureType&SERVICE=WFS
Unable to DescribeFeatureType from URL /qgis/qgis_mapserver.fcgi?map=/var/sig/qgis/distribution.qgs&transparent=true&VERSION=1.1.0&REQUEST=DescribeFeatureType&SERVICE=WFS
ERROR: The database is probably not up to date
(psycopg2.ProgrammingError) relation "main.metadata" does not exist
LINE 2: FROM main.metadata
             ^
 [SQL: 'SELECT main.metadata.id AS main_metadata_id, main.metadata.name AS main_metadata_name, main.metadata.value AS main_metadata_value, main.metadata.description AS main_metadata_descrip                       tion, main.metadata.item_id AS main_metadata_item_id \nFROM main.metadata \nWHERE %(param_1)s = main.metadata.item_id'] [parameters: {'param_1': 677}]
# removes the always changed date line
...............



.build/venv/bin/c2c-template --vars vars_prod_v2.yaml --engine mako --files alembic_static.ini.mako
.build/venv/bin/pot-create --keyword _ --config lingua-server.cfg --output cartoriviera/locale/cartoriviera-server.pot cartoriviera/models.py cartoriviera/templates/mobile.html cartoriviera/templates/api/xapihelp.html cartoriviera/templates/api/apihelp.html cartoriviera/templates/desktop.html cartoriviera/templates/search-tip.html cartoriviera/templates/edit.html cartoriviera/templates/apihelp.html cartoriviera/templates/routing.html
/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/lingua/extract.py:327: UnicodeWarning: Python extractor called with bytes input. Please update your plugin to submit unicode instead.
  for message in extractor(real_filename, options):
# removes the always changed date line
sed -i '/^"POT-Creation-Date: /d' cartoriviera/locale/cartoriviera-server.pot
sed -i '/^"PO-Revision-Date: /d' cartoriviera/locale/cartoriviera-server.pot
chmod go+r cartoriviera/locale/cartoriviera-server.pot
mkdir -p cartoriviera/locale/fr/LC_MESSAGES/
[ -f cartoriviera/locale/fr/LC_MESSAGES/cartoriviera-server.po ] || msginit --no-translator --input=cartoriviera/locale/cartoriviera-server.pot --output-file=cartoriviera/locale/fr/LC_MESSAGES/cartoriviera-server.po -l fr
touch --no-create cartoriviera/locale/fr/LC_MESSAGES/cartoriviera-server.po
msgmerge --backup=none --update --sort-output --no-location cartoriviera/locale/fr/LC_MESSAGES/cartoriviera-server.po cartoriviera/locale/cartoriviera-server.pot
... done.
msgfmt -o cartoriviera/locale/fr/LC_MESSAGES/cartoriviera-server.mo cartoriviera/locale/fr/LC_MESSAGES/cartoriviera-server.po
touch --no-create cartoriviera/locale/fr/LC_MESSAGES/cartoriviera-server.mo
.build/venv/bin/generate_controller --generate-apache-config
Traceback (most recent call last):
  File ".build/venv/bin/generate_controller", line 11, in <module>
    sys.exit(main())
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/tilecloud_chain/controller.py", line 88, in main
    _generate_apache_config(gene)
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/tilecloud_chain/controller.py", line 286, in _generate_apache_config
    with open(gene.config['apache']['config_file'], 'w') as f:
KeyError: 'apache'
CONST_Makefile:1135: recipe for target 'apache/tiles.conf' failed
make[1]: *** [apache/tiles.conf] Error 1
make[1]: Leaving directory '/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish'
Traceback (most recent call last):
  File ".build/venv/bin/c2ctool", line 11, in <module>
    sys.exit(main())
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/c2cgeoportal/scripts/c2ctool.py", line 98, in main
    c2ctool.upgrade()
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/c2cgeoportal/scripts/c2ctool.py", line 222, in upgrade
    self.step4()
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/c2cgeoportal/scripts/c2ctool.py", line 436, in step4
    check_call(["make", "-f", self.options.file, "build"])
  File "/usr/lib/python2.7/subprocess.py", line 540, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['make', '-f', 'v2.mk', 'build']' returned non-zero exit status 2
CONST_Makefile:1237: recipe for target 'upgrade4' failed
make: *** [upgrade4] Error 1

I've added this line at end of config.yaml.mako, and next error is : 

CONST_Makefile:1131: warning: overriding recipe for target 'apache/mapcache.xml'
cartoriviera.mk:17: warning: ignoring old recipe for target 'apache/mapcache.xml'
.build/venv/bin/c2c-template --vars vars_prod_v2.yaml --engine mako --files tilegeneration/config.yaml.mako
.build/venv/bin/generate_controller --generate-mapcache-config
msgfmt -o cartoriviera/locale/fr/LC_MESSAGES/cartoriviera-server.mo cartoriviera/locale/fr/LC_MESSAGES/cartoriviera-server.po
touch --no-create cartoriviera/locale/fr/LC_MESSAGES/cartoriviera-server.mo
.build/venv/bin/generate_controller --generate-apache-config
sudo /usr/sbin/apache2ctl graceful
touch .build/apache.timestamp
.build/venv/bin/c2c-versions CONST_packages.yaml common main print tilecloud-chain deploy ngeo
Usage: {cmd} <config file> <section>
touch .build/test-packages.timestamp
make[1]: Leaving directory '/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish'
INFO  [alembic.runtime.migration] Context impl PostgresqlImpl.
INFO  [alembic.runtime.migration] Will assume transactional DDL.
INFO  [alembic.runtime.migration] Running upgrade daf738d5bae4 -> b60f2a505f42, Remame UIMetadata to Metadata
INFO  [alembic.runtime.migration] Running upgrade b60f2a505f42 -> 8117bb9bba16, Use dimension on all the layers
INFO  [alembic.runtime.migration] Running upgrade 8117bb9bba16 -> e7e03dedade3, Put the default WMS server in the servers part, add some constrains
Traceback (most recent call last):
  File ".build/venv/bin/c2ctool", line 11, in <module>
    sys.exit(main())
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/c2cgeoportal/scripts/c2ctool.py", line 98, in main
    c2ctool.upgrade()
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/c2cgeoportal/scripts/c2ctool.py", line 222, in upgrade
    self.step4()
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/c2cgeoportal/scripts/c2ctool.py", line 438, in step4
    command.upgrade(Config("alembic.ini"), "head")
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/alembic/command.py", line 174, in upgrade
    script.run_env()
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/alembic/script/base.py", line 407, in run_env
    util.load_python_file(self.dir, 'env.py')
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/alembic/util/pyfiles.py", line 93, in load_python_file
    module = load_module_py(module_id, path)
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/alembic/util/compat.py", line 79, in load_module_py
    mod = imp.load_source(module_id, path, fp)
  File "CONST_alembic/main/env.py", line 109, in <module>
    run_migrations_online()
  File "CONST_alembic/main/env.py", line 102, in run_migrations_online
    context.run_migrations()
  File "<string>", line 8, in run_migrations
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/alembic/runtime/environment.py", line 797, in run_migrations
    self.get_context().run_migrations(**kw)
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/alembic/runtime/migration.py", line 312, in run_migrations
    step.migration_fn(**kw)
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/CONST_alembic/main/versions/e7e03dedade3_put_the_default_wms_server_in_the_.py", line 59, in upgrade
    'type_name_unique_treeitem', 'treeitem', ['type', 'name'], schema=schema
  File "<string>", line 8, in create_unique_constraint
  File "<string>", line 3, in create_unique_constraint
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/alembic/operations/ops.py", line 394, in create_unique_constraint
    return operations.invoke(op)
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/alembic/operations/base.py", line 318, in invoke
    return fn(self, operation)
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/alembic/operations/toimpl.py", line 135, in create_constraint
    operation.to_constraint(operations.migration_context)
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/alembic/ddl/impl.py", line 180, in add_constraint
    self._exec(schema.AddConstraint(const))
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/alembic/ddl/impl.py", line 118, in _exec
    return conn.execute(construct, *multiparams, **params)
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 914, in execute
    return meth(self, multiparams, params)
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/sqlalchemy/sql/ddl.py", line 68, in _execute_on_connection
    return connection._execute_ddl(self, multiparams, params)
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 968, in _execute_ddl
    compiled
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1146, in _execute_context
    context)
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1341, in _handle_dbapi_exception
    exc_info
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/sqlalchemy/util/compat.py", line 202, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb, cause=cause)
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 1139, in _execute_context
    context)
  File "/var/www/vhosts/cartoriviera_preprod/private/cartoriviera_geomapfish_v2/cartoriviera_geomapfish/.build/venv/local/lib/python2.7/site-packages/sqlalchemy/engine/default.py", line 450, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.IntegrityError: (psycopg2.IntegrityError) could not create unique index "type_name_unique_treeitem"
DETAIL:  Key (type, name)=(layerv1, orthos_2013) is duplicated.
 [SQL: 'ALTER TABLE main.treeitem ADD CONSTRAINT type_name_unique_treeitem UNIQUE (type, name)']
CONST_Makefile:1237: recipe for target 'upgrade4' failed
make: *** [upgrade4] Error 1

I still have this error with the build : 

Get DescribeFeatureType for url: http://localhost/v2/mapserv?VERSION=1.1.0&REQUEST=DescribeFeatureType&SERVICE=WFS
Get DescribeFeatureType for url: https://www.cartoriviera.ch/qgis/qgis_mapserver.fcgi?MAP=/usr/local/share/qgis/signalisation.qgs&transparent=true&VERSION=1.1.0&REQUEST=DescribeFeatureType&SERVICE=WFS
Get DescribeFeatureType for url: /qgis/qgis_mapserver.fcgi?map=/var/sig/qgis/distribution.qgs&transparent=true&VERSION=1.1.0&REQUEST=DescribeFeatureType&SERVICE=WFS
Unable to DescribeFeatureType from URL /qgis/qgis_mapserver.fcgi?map=/var/sig/qgis/distribution.qgs&transparent=true&VERSION=1.1.0&REQUEST=DescribeFeatureType&SERVICE=WFS
No such table '' for layer 'separateurs_hydro'.
CONST_Makefile:686: recipe for target 'cartoriviera/locale/cartoriviera-client.pot' failed
make: *** [cartoriviera/locale/cartoriviera-client.pot] Error 1

--- a/c2cgeoportal/models.py
+++ b/c2cgeoportal/models.py
@@ -614,6 +614,7 @@ class LayerV1(Layer):  # Deprecated in v2
     id = Column(
         Integer, ForeignKey(_schema + ".layer.id"), primary_key=True
     )
+    layer = Column(Unicode, label=_(u"Layer"))
     is_checked = Column(Boolean, default=True, label=_(u"Checked"))  # by default
     icon = Column(Unicode, label=_(u"Icon"))  # on the tree
     layer_type = Column(Enum(
@@ -661,6 +662,7 @@ class LayerV1(Layer):  # Deprecated in v2
         layer_type=u"internal WMS"
     ):
         Layer.__init__(self, name=name, public=public)
+        self.layer = name
         self.icon = icon
         self.layer_type = layer_type
 

--- a/c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl
+++ b/c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl
@@ -192,7 +192,12 @@ If you want to keep your CGXP interface you should:
 
     example: https://github.com/camptocamp/demo_geomapfish/commit/158c53b5f16c230744f0b101bb3b42a88a0bf969
 
-12. In the database we have added some restrictions, have a look here for more informations:
+12. To pre upgrade the database, run:
+
+    .build/venv/bin/alembic --config alembic.ini upgrade 7d271f4527cd
+    .build/venv/bin/treeitem_uniquename
+
+13. In the database we have added some restrictions, have a look here for more informations:
     https://github.com/camptocamp/c2cgeoportal/issues/2568
 
 

--- a/c2cgeoportal/scaffolds/update/CONST_alembic/main/versions/32b21aa1d0ed_merge_e004f76e951a_and_e004f76e951a_.py
+++ b/c2cgeoportal/scaffolds/update/CONST_alembic/main/versions/32b21aa1d0ed_merge_e004f76e951a_and_e004f76e951a_.py
@@ -1,0 +1,49 @@
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2016, Camptocamp SA
+# All rights reserved.
+
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+# ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+# The views and conclusions contained in the software and documentation are those
+# of the authors and should not be interpreted as representing official policies,
+# either expressed or implied, of the FreeBSD Project.
+
+"""Merge e004f76e951a and e004f76e951a heads
+
+Revision ID: 32b21aa1d0ed
+Revises: ('7d271f4527cd', 'e004f76e951a')
+Create Date: 2016-10-20 15:09:59.215555
+"""
+
+# revision identifiers, used by Alembic.
+revision = '32b21aa1d0ed'
+down_revision = ('7d271f4527cd', 'e004f76e951a')
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    pass
+
+
+def downgrade():
+    pass

--- a/c2cgeoportal/scaffolds/update/CONST_alembic/main/versions/7d271f4527cd_add_layer_column_in_layerv1_table.py
+++ b/c2cgeoportal/scaffolds/update/CONST_alembic/main/versions/7d271f4527cd_add_layer_column_in_layerv1_table.py
@@ -1,0 +1,63 @@
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2016, Camptocamp SA
+# All rights reserved.
+
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+# ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+# The views and conclusions contained in the software and documentation are those
+# of the authors and should not be interpreted as representing official policies,
+# either expressed or implied, of the FreeBSD Project.
+
+"""Add layer column in layerv1 table
+
+Revision ID: 7d271f4527cd
+Revises: 8117bb9bba16
+Create Date: 2016-10-20 15:00:13.619090
+"""
+
+from alembic import op, context
+from sqlalchemy import Column
+from sqlalchemy.types import Unicode
+
+# revision identifiers, used by Alembic.
+revision = '7d271f4527cd'
+down_revision = '8117bb9bba16'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    schema = context.get_context().config.get_main_option('schema')
+
+    op.add_column('layerv1', Column('layer', Unicode), schema=schema)
+    op.execute(
+        'UPDATE {schema}.layerv1 AS l1 '
+        'SET layer = name '
+        'FROM {schema}.treeitem AS ti '
+        'WHERE l1.id = ti.id'.format(schema=schema)
+    )
+
+
+def downgrade():
+    schema = context.get_context().config.get_main_option('schema')
+
+    op.drop_column('layerv1', 'layer', schema=schema)

--- a/c2cgeoportal/scripts/create_demo_theme.py
+++ b/c2cgeoportal/scripts/create_demo_theme.py
@@ -28,23 +28,27 @@
 # either expressed or implied, of the FreeBSD Project.
 
 
+import argparse
 import transaction
-from optparse import OptionParser
 from pyramid.paster import get_app
 
 
 def main():
-    parser = OptionParser("Create and populate the database tables.")
-    parser.add_option(
-        '-i', '--iniconfig', default='production.ini',
+    parser = argparse.ArgumentParser(
+        description="Create and populate the database tables."
+    )
+    parser.add_argument(
+        '-i', '--iniconfig',
+        default='production.ini',
         help='project .ini config file'
     )
-    parser.add_option(
-        '-n', '--app-name', default="app",
+    parser.add_argument(
+        '-n', '--app-name',
+        default="app",
         help='The application name (optional, default is "app")'
     )
 
-    options, args = parser.parse_args()
+    options = parser.parse_args()
 
     # read the configuration
     get_app(options.iniconfig, options.app_name)

--- a/c2cgeoportal/scripts/manage_users.py
+++ b/c2cgeoportal/scripts/manage_users.py
@@ -29,8 +29,8 @@
 
 
 import os.path
+import argparse
 import warnings
-from optparse import OptionParser
 
 from pyramid.paster import get_app
 import transaction
@@ -54,39 +54,45 @@ Reset a user password.
 The username is used as password if the password is not provided with the corresponding option.
 User can be created if it doesn't exist yet."""
 
-    parser = OptionParser(usage)
-    parser.add_option(
-        "-i", "--app-config", default="production.ini",
-        dest="app_config",
+    parser = argparse.ArgumentParser(description=usage)
+    parser.add_argument(
+        "-i", "--app-config",
+        default="production.ini", dest="app_config",
         help="The application .ini config file (optional, default is "
         "'production.ini')"
     )
-    parser.add_option(
-        "-n", "--app-name", default="app", dest="app_name",
+    parser.add_argument(
+        "-n", "--app-name",
+        default="app", dest="app_name",
         help="The application name (optional, default is 'app')"
     )
-    parser.add_option(
+    parser.add_argument(
         "-p", "--password",
         help="Set password (if not set, username is used as password"
     )
-    parser.add_option(
-        "-c", "--create", action="store_true", default=False,
+    parser.add_argument(
+        "-c", "--create",
+        action="store_true", default=False,
         help="Create user if it doesn't already exist"
     )
-    parser.add_option(
-        "-r", "--rolename", default="role_admin",
+    parser.add_argument(
+        "-r", "--rolename",
+        default="role_admin",
         help="The role name which must exist in the database"
     )
-    parser.add_option(
-        "-e", "--email", default=None,
+    parser.add_argument(
+        "-e", "--email",
+        default=None,
         help="The user email"
     )
+    parser.add_argument(
+        'user',
+        nargs='1',
+        help="The user"
+    )
 
-    (options, args) = parser.parse_args()
-    if len(args) != 1:
-        parser.error("You must specify a username")
-
-    username = args[0]
+    options = parser.parse_args()
+    username = options.user
 
     app_config = options.app_config
     app_name = options.app_name

--- a/c2cgeoportal/scripts/treeitem_uniquename.py
+++ b/c2cgeoportal/scripts/treeitem_uniquename.py
@@ -1,0 +1,68 @@
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2014-2016, Camptocamp SA
+# All rights reserved.
+
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+# ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+# The views and conclusions contained in the software and documentation are those
+# of the authors and should not be interpreted as representing official policies,
+# either expressed or implied, of the FreeBSD Project.
+
+
+import argparse
+import transaction
+from pyramid.paster import get_app
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="This script will rename all the theme elements to removes duplicated elements."
+    )
+    parser.add_argument(
+        '-i', '--iniconfig',
+        default='production.ini',
+        help='project .ini config file',
+    )
+    parser.add_argument(
+        '-n', '--app-name',
+        default="app",
+        help='The application name (optional, default is "app")',
+    )
+
+    options = parser.parse_args()
+
+    # read the configuration
+    get_app(options.iniconfig, options.app_name)
+
+    from c2cgeoportal.models import DBSession, LayerV1, LayerWMS, LayerWMTS, LayerGroup, Theme
+
+    for class_ in [LayerV1, LayerWMS, LayerWMTS, LayerGroup, Theme]:
+        names = []
+        for item in DBSession.query(class_).all():
+            if item.name in names:
+                i = 2
+                while "{}-{}".format(item.name, i) in names:
+                    i += 1
+
+                item.name = "{}-{}".format(item.name, i)
+
+    transaction.commit()

--- a/c2cgeoportal/tests/functional/test_entry.py
+++ b/c2cgeoportal/tests/functional/test_entry.py
@@ -929,6 +929,7 @@ class TestEntryView(TestCase):
         layer = LayerV1()
         layer.id = 20
         layer.name = "test internal WMS"
+        layer.layer = "test internal WMS"
         layer.metadata_url = "http://example.com/tiwms"
         layer.is_checked = True
         layer.layer_type = "internal WMS"
@@ -972,6 +973,7 @@ class TestEntryView(TestCase):
         layer = LayerV1()
         layer.id = 20
         layer.name = "test external WMS"
+        layer.layer = "test external WMS"
         layer.is_checked = False
         layer.icon = "static:///tewms.png"
         layer.layer_type = "external WMS"
@@ -1003,6 +1005,7 @@ class TestEntryView(TestCase):
         layer = LayerV1()
         layer.id = 20
         layer.name = "test WMTS"
+        layer.layer = "test WMTS"
         layer.is_checked = False
         layer.layer_type = "WMTS"
         layer.url = "http://example.com/WMTS-Capabilities.xml"
@@ -1038,6 +1041,7 @@ class TestEntryView(TestCase):
         layer = LayerV1()
         layer.id = 20
         layer.name = "test WMTS"
+        layer.layer = "test WMTS"
         layer.is_checked = False
         layer.layer_type = "WMTS"
         layer.url = "http://example.com/WMTS-Capabilities.xml"
@@ -1066,6 +1070,7 @@ class TestEntryView(TestCase):
         layer = LayerV1()
         layer.id = 20
         layer.name = "test WMTS"
+        layer.layer = "test WMTS"
         layer.is_checked = False
         layer.layer_type = "WMTS"
         layer.url = "http://example.com/WMTS-Capabilities.xml"
@@ -1095,6 +1100,7 @@ class TestEntryView(TestCase):
         layer = LayerV1()
         layer.id = 20
         layer.name = "test no 2D"
+        layer.layer = "test no 2D"
         layer.is_checked = False
         layer.layer_type = "no 2D"
         layer.legend = False
@@ -1135,6 +1141,7 @@ class TestEntryView(TestCase):
         layer = LayerV1()
         layer.id = 20
         layer.name = "test_wmsfeaturesgroup"
+        layer.layer = "test_wmsfeaturesgroup"
         layer.layer_type = "internal WMS"
         layer.image_type = "image/png"
         layer.is_checked = False
@@ -1170,6 +1177,7 @@ class TestEntryView(TestCase):
         layer_t1 = LayerV1()
         layer_t1.id = 20
         layer_t1.name = "test_wmstime"
+        layer_t1.layer = "test_wmstime"
         layer_t1.layer_type = "internal WMS"
         layer_t1.image_type = "image/png"
         layer_t1.is_checked = False
@@ -1193,6 +1201,7 @@ class TestEntryView(TestCase):
         layer_t2 = LayerV1()
         layer_t2.id = 30
         layer_t2.name = "test_wmstime2"
+        layer_t2.layer = "test_wmstime2"
         layer_t2.layer_type = "internal WMS"
         layer_t2.image_type = "image/png"
         layer_t2.is_checked = False
@@ -1236,6 +1245,7 @@ class TestEntryView(TestCase):
         layer = LayerV1()
         layer.id = 20
         layer.name = "test_wmstimegroup"
+        layer.layer = "test_wmstimegroup"
         layer.layer_type = "internal WMS"
         layer.image_type = "image/png"
         layer.is_checked = False
@@ -1260,6 +1270,7 @@ class TestEntryView(TestCase):
         layer = LayerV1()
         layer.id = 20
         layer.name = "test WMTS"
+        layer.layer = "test WMTS"
         layer.is_checked = False
         layer.layer_type = "WMTS"
         layer.url = "http://example.com/WMTS-Capabilities.xml"
@@ -1289,6 +1300,7 @@ class TestEntryView(TestCase):
         layer = LayerV1()
         layer.id = 20
         layer.name = "test WMTS"
+        layer.layer = "test WMTS"
         layer.is_checked = False
         layer.layer_type = "WMTS"
         layer.url = "http://example.com/WMTS-Capabilities.xml"
@@ -1326,6 +1338,7 @@ class TestEntryView(TestCase):
         layer = LayerV1()
         layer.id = 20
         layer.name = "test layer in group"
+        layer.layer = "test layer in group"
         layer.is_checked = False
         layer.layer_type = "internal WMS"
         layer.image_type = "image/png"

--- a/c2cgeoportal/views/entry.py
+++ b/c2cgeoportal/views/entry.py
@@ -357,7 +357,7 @@ class Entry:
         errors = set()
         l = {
             "id": layer.id,
-            "name": layer.name,
+            "name": layer.name if not isinstance(layer, LayerV1) else layer.layer,
             "metadata": self._get_metadatas(layer, errors),
         }
         if re.search("[/?#]", layer.name):  # pragma: no cover

--- a/doc/developer/server_side.rst
+++ b/doc/developer/server_side.rst
@@ -318,11 +318,7 @@ Or in c2cgeoportal root directory:
         --config c2cgeoportal/tests/functional/alembic[_static].ini \
         revision --message "<Explicit name>"
 
-Where ``<branch>`` is the branch on witch we will merge, usually ``master`` but
-it can also be something like ``2.0``.
-
-This will generate the migration script in
-``CONST_alembic/versions/xxx_<Explicite_name>.py``
+This will generate the migration script in ``CONST_alembic/versions/xxx_<Explicite_name>.py``
 You should *NOT* commit the script in this directory because this migration
 script should be shared with all c2cgeoportal projects.
 It is the c2cgeoportal ``update`` template which is responsible for updating

--- a/setup.py
+++ b/setup.py
@@ -110,6 +110,7 @@ setup(
             "gen-version = c2cgeoportal.scripts.gen_version:main",
             "ngeo-version = c2cgeoportal.scripts.import_ngeo_apps:ngeo_git_version",
             "create-demo-theme = c2cgeoportal.scripts.create_demo_theme:main",
+            "treeitem_uniquename = c2cgeoportal.scripts.treeitem_uniquename:main",
         ],
         "pyramid.scaffold": [
             "c2cgeoportal_create = c2cgeoportal.scaffolds:TemplateCreate",

